src/components/AutoSuggest.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react';
+import Autosuggest, { SuggestionSelectedEventData } from 'react-autosuggest';
+
+import {
+	useStyles,
+	OptionType,
+	getSuggestionValue,
+	renderInputComponent,
+	renderSuggestion,
+	performSearch
+} from './ReactAutoSuggestHelper';
+import { Paper } from '@material-ui/core';
+
+interface AutoSuggestProps {
+	label: string;
+	onChange: (value: string) => void;
+	onClick?: (id: string) => void;
+	placeholder?: string;
+	registryName: string;
+	value: string;
+}
+
+const AutoSuggest: React.FC<AutoSuggestProps> = ({
+	label,
+	onChange,
+	onClick,
+	placeholder,
+	registryName,
+	value
+}) => {
+	const classes = useStyles();
+
+	const [inputText, setInputText] = useState(value || '');
+	const [suggestions, setSuggestions] = React.useState<OptionType[]>([]);
+
+	const handleSuggestionsFetchRequested = ({ value }: any) => {
+		onChange(value);
+		if (value.length > 2) {
```

Comment:
The magic number two (here setting the number of chars before search is performed?) should preferably be a constant of some kind — possibly a parameter for the app